### PR TITLE
 feat(payments): coupon Subscription Confirmation Page

### LINKED ISSUE #11470: Update Subscription Confirmation Page to Reflect Coupon Amount (Old and New Flow)
Per the figma mockups, the subscription confirmation page needs to show the coupon data in the plan details component on that page and needs to have the payment information reflect the discounted amount.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-4370)

### PR DESCRIPTION
**Pull request by IvoJP**
Because:

* We want the subscription confirmation page to reflect coupon discounts

This commit:

* Adds the coupon/discount data to the subscriptionSuccess component

Closes #11470 

_Put an `x` in the boxes that apply_

- [X] My commit is GPG signed.
- [X] If applicable, I have modified or added tests which pass locally.
- [X] I have added necessary documentation (if appropriate).
- [X] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx [hunk 1]
```diff
@@ -53,7 +56,7 @@ export const PaymentConfirmation = ({
   });
 
   const planPrice = formatPlanPricing(
-    amount,
+    coupon && amount ? amount - coupon.amount : amount,
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx:59]: Maybe capture this into a variable?  Line 139 below is doing the same thing.

File: packages/fxa-payments-server/src/routes/Checkout/index.tsx [hunk 2]
```diff
@@ -243,6 +243,7 @@ export const Checkout = ({
       ]);
       setProfile(profile);
       setCustomer(customer);
+      setCoupon(coupon);
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Checkout/index.tsx:246]: I don't understand this call.

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx [hunk 3]
```diff
@@ -71,6 +71,8 @@ export type SubscriptionCreateProps = {
   stripeOverride?: SubscriptionCreateStripeAPIs;
   apiClientOverrides?: Partial<SubscriptionCreateAuthServerAPIs>;
   paypalButtonBase?: React.FC<ButtonBaseProps>;
+  coupon?: Coupon;
+  setCoupon: (value: any) => void;
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx:75]: Isn't the value a `Coupon`?

**Review by chenba — CHANGES_REQUESTED** [review]
Looks good.  But there's a `setCoupon` call I don't understand.  The coupon's state hasn't changed at that point afaict.

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx (hunk 3 above)
Comment (IvoJP) [reply packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx]: yes

File: packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx (hunk 1 above)
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx]: done

File: packages/fxa-payments-server/src/routes/Checkout/index.tsx (hunk 2 above)
Comment (IvoJP) [reply packages/fxa-payments-server/src/routes/Checkout/index.tsx]: left over from development - its unnecessary. removed.

**Pull request merged**